src/application/shell.ts

```diff
@@ -280,7 +280,7 @@ class ApplicationShell extends Widget {
    * Close all tracked widgets.
    */
   closeAll(): void {
-    each(this._dockPanel.widgets, widget => { widget.close(); });
+    each(this._dockPanel.widgets(), widget => { widget.close(); });
```

Comment:
This is not safe. Closing a widget removes it from the dock panel, which invalidates the `widgets()` iterator. You should exhaust the iterator into an array before calling `close` on the widgets.